apps/mochi-web/components/settings/general/TransactionLimit.tsx

```diff
@@ -0,0 +1,211 @@
+  Control,
+  Controller,
+  UseFormWatch,
+  useFieldArray,
+} from 'react-hook-form'
+import clsx from 'clsx'
+import { actionList } from '~constants/settings'
+import { utils as mochiUtils } from '@consolelabs/mochi-ui'
+import { GeneralFormValue } from './types'
+import { TransactionLimitModal } from './TransactionLimitModal'
+
+const tokenList = [
+  {
+    id: '941f0fb1-00da-49dc-a538-5e81fc874cb4',
+    name: 'Icy',
+    symbol: 'ICY',
+    decimal: 18,
+    chain_id: '137',
+    native: false,
+    address: '0x8D57d71B02d71e1e449a0E459DE40473Eb8f4a90',
+    icon: 'https://cdn.discordapp.com/emojis/1049620715374133288.webp?size=240&quality=lossless',
+    coin_gecko_id: 'icy',
+    price: 1.5,
+    chain: {
+      id: '7303f2f8-b6d9-454d-aa92-880569fa5295',
+      chain_id: '137',
+      name: 'Polygon Mainnet',
+      symbol: 'MATIC',
+      rpc: 'https://polygon.llamarpc.com',
+      explorer: 'https://polygonscan.com',
+      icon: 'https://cdn.discordapp.com/emojis/928216430535671818.png?size=240&quality=lossless',
+      type: 'evm',
+    },
+  },
+  {
+    id: 'd0bf0f44-e951-4f81-8643-3e6b9b0841d8',
+    name: 'Ethereum',
+    symbol: 'ETH',
+    decimal: 18,
+    chain_id: '42161',
+    native: true,
+    address: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
+    icon: '',
+    coin_gecko_id: 'ethereum',
+    price: 2129.75,
+    chain: {
+      id: 'b1065911-fd00-4424-bc7c-03868e6d1ed1',
+      chain_id: '42161',
+      name: 'Arbitrum',
+      symbol: 'ARB',
+      rpc: 'https://arbitrum.llamarpc.com',
+      explorer: 'https://arbiscan.io',
+      icon: '',
+      type: 'evm',
+    },
+  },
+  {
+    id: '61388b7c-5505-4fdf-8084-077422369a93',
+    name: 'Fantom',
+    symbol: 'FTM',
+    decimal: 18,
+    chain_id: '250',
+    native: true,
+    address: '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
+    icon: '',
+    coin_gecko_id: 'fantom',
+    price: 0.380276,
+    chain: {
+      id: '7303f2f8-b6d9-454d-aa92-880569fa5296',
+      chain_id: '250',
+      name: 'Fantom Opera',
+      symbol: 'FTM',
+      rpc: 'https://rpc.ftm.tools',
+      explorer: 'https://ftmscan.com',
+      icon: 'https://cdn.discordapp.com/emojis/928216448902508564.png?size=240&quality=lossless',
+      type: 'evm',
+    },
+  },
+]
+
+interface Props {
+  control: Control<GeneralFormValue>
+  watch: UseFormWatch<GeneralFormValue>
+}
+
+export const TransactionLimit = ({ control, watch }: Props) => {
+  const { fields, append, remove, update } = useFieldArray({
+    control,
+    name: 'transactionLimit',
+  })
+  const enableTransactionLimit = watch('enableTransactionLimit')
+
+  return (
+    <div className="flex flex-col w-full max-w-md space-y-2">
+      <div className="flex items-center justify-between">
+        <FormLabel>Set the limit for transaction</FormLabel>
+        <Controller
+          name="enableTransactionLimit"
+          control={control}
+          render={({ field: { value, onChange, ...rest } }) => (
+            <Switch
```

Comment:
Hide the content instead of blurring it out

<img width="563" alt="image" src="https://github.com/consolelabs/mochi-ui/assets/12707960/b01785c3-8437-4e7f-a487-db724571e1a0">